app/assets/javascripts/src/Metamaps.GlobalUI.js

```diff
@@ -43,31 +43,45 @@ Metamaps.Active = {
 Metamaps.Maps = {};
 
 $(document).ready(function () {
-  function init() {
-    for (var prop in Metamaps) {
-
-        // this runs the init function within each sub-object on the Metamaps one
-        if (Metamaps.hasOwnProperty(prop) &&
-            Metamaps[prop] != null &&
-            Metamaps[prop].hasOwnProperty('init') &&
-            typeof (Metamaps[prop].init) == 'function'
-        ) {
-            Metamaps[prop].init();
-        }
-    }
+  // initialize all the modules
+  for (var prop in Metamaps) {
+      // this runs the init function within each sub-object on the Metamaps one
+      if (Metamaps.hasOwnProperty(prop) &&
+          Metamaps[prop] != null &&
+          Metamaps[prop].hasOwnProperty('init') &&
+          typeof (Metamaps[prop].init) == 'function'
+      ) {
+          Metamaps[prop].init();
+      }
+  }
+  // load whichever page you are on
+  if (Metamaps.currentSection === "explore") {
+      var capitalize = Metamaps.currentPage.charAt(0).toUpperCase() + Metamaps.currentPage.slice(1);
+
+      Metamaps.Views.exploreMaps.setCollection( Metamaps.Maps[capitalize] );
+      if (Metamaps.currentPage === "mapper") {
+          Metamaps.Views.exploreMaps.fetchUserThenRender();
+          Metamaps.Header.fetchUserThenChangeSection(!!Metamaps.Active.Mapper, Metamaps.Maps.Mapper.mapperId)
+      }
+      else {
```

Comment:
Perform the page initialization that was happening in the Famous code in the main thread, after everythings been initialized